### PR TITLE
Docs: Lifecycle Management of Counters and Lists in REST

### PR DESCRIPTION
**Pull request by Kalaiselvi84**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, press enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking
> /kind bug
> /kind cleanup

/kind documentation
> /kind feature
> /kind hotfix
> /kind release

**What this PR does / Why we need it**:

**Which issue(s) this PR fixes**:

Closes #

**Special notes for your reviewer**:

### CODE REVIEW

File: site/content/en/docs/Guides/Client SDKs/rest.md
Comment (igooch) [inline site/content/en/docs/Guides/Client SDKs/rest.md:251]: We should probably mention in the example that conformanceTestCounter is predefined in the local SDK Server, or go with the more generic example of {name} or {counterName}.

File: site/content/en/docs/Guides/Client SDKs/rest.md
Comment (igooch) [inline site/content/en/docs/Guides/Client SDKs/rest.md:291]:
```suggestion
This function updates the specified list's properties, such as its capacity and values, returns the updated list details. This will overwrite all existing List.Values with the update list request values. Use addValue or removeValue for modifying the List.Values field.
```

**Review by igooch — COMMENTED** [review]
Looks good!

File: site/content/en/docs/Guides/Client SDKs/rest.md
Comment (Kalaiselvi84) [reply site/content/en/docs/Guides/Client SDKs/rest.md:266]: I have included the Local Development Setup section for Counters and Lists in this document. Could you please review it and share your thoughts?

File: site/content/en/docs/Guides/Client SDKs/rest.md
Comment (markmandel) [inline site/content/en/docs/Guides/Client SDKs/rest.md:254]:
```suggestion
```

Delete all this please - this page isn't covering the local SDK usage, just the REST API.

File: site/content/en/docs/Guides/Client SDKs/rest.md
Comment (markmandel) [inline site/content/en/docs/Guides/Client SDKs/rest.md:241]:
```suggestion
### Counters and Lists
```

File: site/content/en/docs/Guides/Client SDKs/rest.md
Comment (markmandel) [reply site/content/en/docs/Guides/Client SDKs/rest.md:254]: As a note - we could make a section in https://agones.dev/site/docs/guides/client-sdks/local/ on how to enable feature flags (which we have above), which would be useful! @igooch wdyt?

File: site/content/en/docs/Guides/Client SDKs/rest.md
Comment (markmandel) [reply site/content/en/docs/Guides/Client SDKs/rest.md:266]: I would add an explainer under each example that uses a key, something like:

> In this example, we retrieve the counter under the key "conformanceTestCounter" to make it extra clear.

Since people don't always read the top level docs 😃

As an improvement to the docs and likely the local sdk experience,  @igooch I didn't catch this earlier, but any objection to changing:
https://github.com/googleforgames/agones/blob/1e78d88a27ae7d427e43f073aab7d28289b7d850/pkg/sdkserver/localsdk.go#L147-L158

To using some more generic keys? Maybe `players` for the list and `rooms` for the count? 

We don't actually use these values for conformance tests anyway, so this might be a bit confusing, and I think would make examples easier to understand. WDYT? If so, we can do this in a separate PR (and will need to feature shortcode the new docs) - if you like we can file a new issue.

File: site/content/en/docs/Guides/Client SDKs/rest.md
Comment (Kalaiselvi84) [reply site/content/en/docs/Guides/Client SDKs/rest.md]: I've added sections for `Enabling Specific Features` and `Setting the HTTP port`.

File: site/content/en/docs/Guides/Client SDKs/rest.md
Comment (Kalaiselvi84) [reply site/content/en/docs/Guides/Client SDKs/rest.md:253]: >I would add an explainer under each example that uses a key, something like:
In this example, we retrieve the counter under the key "conformanceTestCounter" to make it extra clear.

I have included this information in a `Note` section located under both the `Counters` and `Lists` titles, following the `Warning` section. This will avoid the repetition right? WDYT?

>To using some more generic keys? Maybe players for the list and rooms for the count?

Do you want to replace `"conformanceTestCounter": {Count: 1, Capacity: 10}} ` with `"rooms": {Count: 1, Capacity: 10}} ` and `"conformanceTestList": {Values: []string{"test0", "test1", "test2"}, Capacity: 100}}` with `"players": {Values: []string{"player1", "player2", "player3"}, Capacity: 100}}`?

File: site/content/en/docs/Guides/Client SDKs/rest.md
Comment (Kalaiselvi84) [reply site/content/en/docs/Guides/Client SDKs/rest.md]: I haven't included it. Is it really necessary? 🤔

File: site/content/en/docs/Guides/Client SDKs/local.md
Comment (markmandel) [inline site/content/en/docs/Guides/Client SDKs/local.md:219]:
```suggestion
```

This isn't accurate for the local SDK server, it's either --http-port argument, or HTTP_PORT (similar for grpc). Since port changing isn't that common, I'd suggest not worrying about documenting this -- unless anyone feels strongly otherwise?

If people really want it, they can run `--help` on the local server as well to get all the command line params.

File: site/content/en/docs/Guides/Client SDKs/local.md
Comment (markmandel) [inline site/content/en/docs/Guides/Client SDKs/local.md:216]: Let's create move this right after the section "Running In “Local Mode".

````suggestion

## Enabling Feature Gates

For development and testing purposes, you might want to enable specific [features gates]({{< relref "../feature-stages.md#feature-gates" >}}) in the local SDK Server. 

To do this, you can either set the `FEATURE_GATES` environment variable or use the `--feature-gates` command line parameter like so, with the same format as utilised when [configuring it on a Helm install]({{< ref "/docs/Installation/Install Agones/helm.md#configuration" >}}).

For example:

```bash
./sdk-server.linux.amd64 --local --feature-gates Example=true
```
or 
```bash
FEATURE_GATES=Example=true ./sdk-server.linux.amd64 --local
```


````

You'll probably have to fix my link paths though - I did this off the top of my head.

File: site/content/en/docs/Guides/Client SDKs/rest.md
Comment (markmandel) [inline site/content/en/docs/Guides/Client SDKs/rest.md:243]:
```suggestion
{{< alpha title="Counters and Lists" gate="CountsAndLists" >}}
```

There is only one feature flag for both Counters and Lists - so it needs to be both - hence I left in the top level header of "Counters and Lists" over each "Counters" and "Lists" heading.

File: site/content/en/docs/Guides/Client SDKs/rest.md
Comment (markmandel) [reply site/content/en/docs/Guides/Client SDKs/rest.md:253]: > We don't actually use these values for conformance tests anyway

I'm silly - of course we use this for conformance tests 🤦🏻 . Ignore that comment. But I am wondering if we change the key anyway, to make it easier for end users, and make our conformance testing match. WDYT?

For these docs, we don't have to conform to the local SDK for the example, so I think we can change the key to something more generic like I suggest above. I'd like to get agreement on the code changes from @igooch before we make the code changes though to the local sdk server (and either way, we should handle that in a separate PR).

Sound good?

File: site/content/en/docs/Guides/Client SDKs/rest.md
Comment (markmandel) [inline site/content/en/docs/Guides/Client SDKs/rest.md:253]:
```suggestion
curl -H "Content-Type: application/json" -X GET http://localhost:${AGONES_SDK_HTTP_PORT}/v1alpha1/counters/rooms
```

(and all the other references)

File: site/content/en/docs/Guides/Client SDKs/rest.md
Comment (markmandel) [inline site/content/en/docs/Guides/Client SDKs/rest.md:245]:
```suggestion
In all the  Counter examples, we retrieve the counter under the key `rooms` as if it was previously defined in [`GameServer.Spec.counters[room]`]({{< ref "/docs/reference/agones_crd_api_reference/#agones.dev/v1.GameServerSpec" >}})

For your own Counter REST requests, replace the value `rooms` with your own key in the path.
```

File: site/content/en/docs/Guides/Client SDKs/rest.md
Comment (markmandel) [inline site/content/en/docs/Guides/Client SDKs/rest.md:279]:
```suggestion
In all the List examples, we retrieve the list under the key `players` as if it was previously defined in [`GameServer.Spec.lists[players]`]({{< ref "/docs/reference/agones_crd_api_reference/#agones.dev/v1.GameServerSpec" >}})

For your own List REST based requests, replace the value `players` with your own key in the path.
```

File: site/content/en/docs/Guides/Client SDKs/rest.md
Comment (markmandel) [inline site/content/en/docs/Guides/Client SDKs/rest.md:291]:
```suggestion
{"name":"players", "capacity":"100", "values":["player0", "player1", "player2"]}     
```

And same as other `conformanceTestList` references below.

File: site/content/en/docs/Guides/Client SDKs/local.md
Comment (Kalaiselvi84) [reply site/content/en/docs/Guides/Client SDKs/local.md]: I've eliminated this `Setting the HTTP port:` section

File: site/content/en/docs/Guides/Client SDKs/local.md
Comment (Kalaiselvi84) [reply site/content/en/docs/Guides/Client SDKs/local.md]: Moved `Enabling Feature Gates` section to immediately follow the `Running In “Local Mode"`

File: site/content/en/docs/Guides/Client SDKs/rest.md
Comment (Kalaiselvi84) [reply site/content/en/docs/Guides/Client SDKs/rest.md]: Included the header.

File: site/content/en/docs/Guides/Client SDKs/rest.md
Comment (Kalaiselvi84) [reply site/content/en/docs/Guides/Client SDKs/rest.md]: Renamed to `rooms` in all the references

File: site/content/en/docs/Guides/Client SDKs/rest.md
Comment (Kalaiselvi84) [reply site/content/en/docs/Guides/Client SDKs/rest.md]: done!

File: site/content/en/docs/Guides/Client SDKs/rest.md
Comment (Kalaiselvi84) [reply site/content/en/docs/Guides/Client SDKs/rest.md]: Renamed to `players` in all the references.

File: site/content/en/docs/Guides/Client SDKs/rest.md
Comment (Kalaiselvi84) [reply site/content/en/docs/Guides/Client SDKs/rest.md]: changed it as per your suggestion

**Comment by Kalaiselvi84** [comment]
I think I've made all the changes you suggested.👍🏻

File: site/content/en/docs/Guides/Client SDKs/local.md
Comment (markmandel) [inline site/content/en/docs/Guides/Client SDKs/local.md:61]: This doesn't need to be feature gated - it's applicable everywhere., otherwise, this looks great!

File: site/content/en/docs/Guides/Client SDKs/rest.md
Comment (markmandel) [inline site/content/en/docs/Guides/Client SDKs/rest.md:241]: No need to feature gate these docs either, since it's already been released 👍🏻

File: site/content/en/docs/Guides/Client SDKs/rest.md
Comment (markmandel) [inline site/content/en/docs/Guides/Client SDKs/rest.md:246]:
```suggestion
#### Counters
```

Just so it's semantically under "Counters and Lists"

File: site/content/en/docs/Guides/Client SDKs/rest.md
Comment (markmandel) [inline site/content/en/docs/Guides/Client SDKs/rest.md:252]:
```suggestion
##### Alpha: GetCounter
```

Same for the other headers under `#### Counters`

File: site/content/en/docs/Guides/Client SDKs/rest.md
Comment (markmandel) [inline site/content/en/docs/Guides/Client SDKs/rest.md:280]:
```suggestion
#### Lists
```

File: site/content/en/docs/Guides/Client SDKs/rest.md
Comment (markmandel) [inline site/content/en/docs/Guides/Client SDKs/rest.md:286]:
```suggestion
##### Alpha: GetList
```

and again, same as below.

**Review by markmandel — COMMENTED** [review]
Some minor pieces, but otherwise, LGTM! Also waiting on approval from @igooch as well 👍🏻

**Comment by Kalaiselvi84** [comment]
All the changes are in place.👍🏻

File: site/content/en/docs/Guides/Client SDKs/local.md
Comment (igooch) [inline site/content/en/docs/Guides/Client SDKs/local.md]:
```suggestion
```

File: site/content/en/docs/Guides/Client SDKs/rest.md
Comment (igooch) [inline site/content/en/docs/Guides/Client SDKs/rest.md]:
```suggestion
```

**Review by igooch — COMMENTED** [review]
Looks good, just a +1 to Mark's comments on not needing to wait until the next release.

File: site/content/en/docs/Guides/Client SDKs/rest.md
Comment (igooch) [reply site/content/en/docs/Guides/Client SDKs/rest.md]: My personal preference is to have the example match what we have in the localSDKServer for a better user experience. That way the user can run the example code and it works without them needing to make changes. My general gripe with generic examples is that they implicitly assume the user knows what and where to make changes without any accompanying instructions. That being said the key "conformanceTestCounter" probably doesn't mean anything to the user either, so I'd rename if along the lines of "TestCounterKey", "TestPlayerCounter", etc.

I'm also fine leaving the examples more generic here, as long as we have a step by step guide elsewhere that is more accessible to the user that needs more context on how and where to make changes.

File: site/content/en/docs/Guides/Client SDKs/local.md
Comment (Kalaiselvi84) [reply site/content/en/docs/Guides/Client SDKs/local.md]: 🤔I have removed the feature shortcode in the [latest commit](https://github.com/googleforgames/agones/pull/3560/commits/561f1e4e693fa79a966e4f675311daadc08dadbf_

Comment (igooch) [reply site/content/en/docs/Guides/Client SDKs/local.md]: Ah yep, looks like you added the new commit before I finished my review. Looks good!

File: site/content/en/docs/Guides/Client SDKs/rest.md
Comment (markmandel) [reply site/content/en/docs/Guides/Client SDKs/rest.md]: > My personal preference is to have the example match what we have in the localSDKServer for a better user experience.

100% agreed - no disagreement there - sounds like we're just talking about what names to give keys. This change I've suggested above I would like to see as an interim step until we can align it again next release with a future PR to update the local sdk server and conformance tests.

> That being said the key "conformanceTestCounter" probably doesn't mean anything to the user either, so I'd rename the keys along the lines of "TestCounterKey", "TestPlayerCounter", etc.

So i would argue that both "test", "key", and "counter" are redundant in the name, and just confuse things a bit more, and it's easier to give a real world use case for the local SDK user that is easy to understand (and will align with the eventual integration pattern docs I'll write - primarily one for "rooms" and one for "player tracking" - as well as the examples we have in the yaml files etc.).

That being said, I would be super happy if as part of the description for each method, there is a reminder for what the key is that is being used, since reminders are great and redundancy in docs is definitely not a bad thing, I'll write up a small suggestion, lemme know what you think.

File: site/content/en/docs/Guides/Client SDKs/rest.md
Comment (Kalaiselvi84) [reply site/content/en/docs/Guides/Client SDKs/rest.md:266]: Would it be a good idea if we rephrase like this?
This function updates the properties of the counter with the key `rooms`, such as its count and capacity, and returns the updated counter details.

Comment (markmandel) [reply site/content/en/docs/Guides/Client SDKs/rest.md:266]: That reads better! Nice.

File: site/content/en/docs/Guides/Client SDKs/rest.md
Comment (Kalaiselvi84) [reply site/content/en/docs/Guides/Client SDKs/rest.md]: Are we sticking with `rooms` and `players` for now and considering changes later if needed?

File: site/content/en/docs/Guides/Client SDKs/rest.md
Comment (Kalaiselvi84) [reply site/content/en/docs/Guides/Client SDKs/rest.md]: I've made the suggested changes. Ivy, wdyt?

**Review by igooch — APPROVED** [review]
LGTM

File: site/content/en/docs/Guides/Client SDKs/rest.md
Comment (markmandel) [inline site/content/en/docs/Guides/Client SDKs/rest.md:299]: Can we please do `players` in this List section like we did for `rooms` above, and then I think this PR is good to go.

```suggestion
This function updates the list's properties with the key `players`, such as its capacity and values, returns the updated list details. This will overwrite all existing List.Values with the update list request values. Use addValue or removeValue for modifying the List.Values field.
```

File: site/content/en/docs/Guides/Client SDKs/rest.md
Comment (markmandel) [inline site/content/en/docs/Guides/Client SDKs/rest.md:313]:
```suggestion
This function adds a new value to a list with the key `players` and returns the list with this addition.
```

File: site/content/en/docs/Guides/Client SDKs/rest.md
Comment (markmandel) [inline site/content/en/docs/Guides/Client SDKs/rest.md:327]:
```suggestion
This function removes a value from the list with the key `players` and returns updated list.
```

**Review by markmandel — COMMENTED** [review]
Almost there!

**Review by markmandel — APPROVED** [review]

**Pull request merged**
